pkg/pget.go

```diff
@@ -0,0 +1,52 @@
+package pget
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/dustin/go-humanize"
+
+	"github.com/replicate/pget/pkg/consumer"
+	"github.com/replicate/pget/pkg/download"
+	"github.com/replicate/pget/pkg/logging"
+)
+
+type Getter struct {
+	Downloader download.Strategy
+	Consumer   consumer.Consumer
+}
+
+func (g *Getter) DownloadFile(ctx context.Context, url string, dest string) (int64, time.Duration, error) {
+	if g.Consumer == nil {
+		g.Consumer = &consumer.FileWriter{}
+	}
+	logger := logging.GetLogger()
+	downloadStartTime := time.Now()
+	buffer, fileSize, err := g.Downloader.Fetch(ctx, url)
+	if err != nil {
+		return fileSize, 0, err
+	}
+	writeStartTime := time.Now()
+	downloadElapsed := time.Since(downloadStartTime)
+
+	err = g.Consumer.Consume(buffer, dest)
+	if err != nil {
+		return fileSize, 0, fmt.Errorf("error writing file: %w", err)
+	}
+	writeElapsed := time.Since(writeStartTime)
+	totalElapsed := time.Since(downloadStartTime)
+
+	size := humanize.Bytes(uint64(fileSize))
+	throughput := humanize.Bytes(uint64(float64(fileSize) / float64(totalElapsed)))
+	logger.Debug().
+		Str("dest", dest).
+		Str("size", size).
+		Str("overall_throughput", throughput).
+		Str("download_elapsed", fmt.Sprintf("%.3fs", downloadElapsed.Seconds())).
+		Str("write_elapsed", fmt.Sprintf("%.3fs", writeElapsed.Seconds())).
+		Str("total_elapsed", fmt.Sprintf("%.3fs", totalElapsed.Seconds())).
+		Msg("Write Complete")
```

Comment:
Let's make the logging in buffer.go be a more useful Debug (if anything) output, let's convert this one over to the .Info(), same with the writer interfaces. I think we should have interesting debug (if anything) there and consolidate the logging here for the info output. Maybe this is too much info now? 

Note: I switched the writeStartTime to be after the download complete 

First Option:
```suggestion
	downloadElapsed := time.Since(downloadStartTime)
	writeStartTime := time.Now()
	
	err = g.Consumer.Consume(buffer, dest)
	if err != nil {
		return fileSize, 0, fmt.Errorf("error writing file: %w", err)
	}
	writeElapsed := time.Since(writeStartTime)
	totalElapsed := time.Since(downloadStartTime)

	size := humanize.Bytes(uint64(fileSize))
	downloadThroughput := humanize.Bytes(uint64(float64(fileSize) / float64(downloadElapsed)))
	writeThroughput := humanize.Bytes(uint64(float64(fileSize) / float64(writeElapsed)))
 logger.Info().
		Str("dest", dest).
		Str("size", size).
		Str("write_throughput", writeThrougput).
		Str("download_throughput", downloadThroughput).
		Str("download_elapsed", fmt.Sprintf("%.3fs", downloadElapsed.Seconds())).
		Str("write_elapsed", fmt.Sprintf("%.3fs", writeElapsed.Seconds())).
		Str("total_elapsed", fmt.Sprintf("%.3fs", totalElapsed.Seconds())).
		Msg("Complete")
```
Alternate option:
keep super relevant (size, dest, total_elapsed, (download_throughput? since it was in the original output of pget) here in .Info() and add other things like write_elapsed and download_elapsed into an explicit .Debug() log call?